### PR TITLE
Add in middleware healthchecks

### PR DESCRIPTION
**Pull request by thegreatercurve**
- Add in Pingdom healthchecks for each of the URLs used in the middlewares that throw 401s. 
 - Bump up the threshold for the standard 401 checks. 
 - Refactor and abstract a util method in the middlewares.

[Ticket](https://financialtimes.atlassian.net/browse/NOPSCOPS-224?atlOrigin=eyJpIjoiYWZkNzA0MTg3ZWE5NDM1YmExYTAxNzVlZmM4M2EwOGYiLCJwIjoiaiJ9)

### CODE REVIEW

File: health/middlewares.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const nHealth = require('n-health');
+
+const { HEALTCHECK_DUMMY_USER_UUID } = require('config');
```
Comment (binaryberry) [inline health/middlewares.js:3]: Where does this config live?

File: health/middlewares.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		name: 'Syndication user rights reachable',
+		url: 'https://session-next.ft.com/products',
+		technicalSummary:
+			'This middleware tests that a user has a valid syndication licence by logging in a dummy user.'
+	},
+	{
+		name: 'Syndication licence reachable',
+		url: `https://api.ft.com/licences?userid=${HEALTCHECK_DUMMY_USER_UUID}`,
+		technicalSummary:
+			'This middleware does the GET the syndication licence details by passing in the unique User ID.'
```
Comment (umbobabo) [inline health/middlewares.js:19]:
```suggestion
			'Endpoint to retrieve the syndication licence details by passing a unique User ID.'
```

File: health/middlewares.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	},
+	{
+		name: 'User profile reachable',
+		url: `https://api.ft.com/users/${HEALTCHECK_DUMMY_USER_UUID}/profile/basic`,
+		technicalSummary:
+			'This middleware does the GET to get the user profile details by passing the unique User ID.'
```
Comment (umbobabo) [inline health/middlewares.js:25]:
```suggestion
			'Endpoint to get the user profile details by passing a unique User ID.'
```

File: health/middlewares.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+const nHealth = require('n-health');
+
+const { HEALTCHECK_DUMMY_USER_UUID } = require('config');
+
+const panicGuide = 'Contact the FT Accounts team #ft-accounts-team.';
+const businessImpact = 'Users will be unable to use the Syndication apps';
+
+const middlewares = [
```
Comment (umbobabo) [inline health/middlewares.js:8]: I think middlewares is a bit of misleading term here. I think you should mentions the following as `Services` and related endpoints. [See also here](https://github.com/Financial-Times/next-profile/blob/b62423a62d612c1b49c2ef27d330ce79ff71906d/server/config/health-checks/membership.js#L136).

File: health/middlewares.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		name: 'User profile reachable',
+		url: `https://api.ft.com/users/${HEALTCHECK_DUMMY_USER_UUID}/profile/basic`,
+		technicalSummary:
+			'This middleware does the GET to get the user profile details by passing the unique User ID.'
+	},
+	{
+		name: 'User authentication reachable',
+		url: 'https://api.ft.com/authorize',
+		technicalSummary:
+			'This middleware does the GET to get token required to view a user\'s profile details.'
```
Comment (umbobabo) [inline health/middlewares.js:31]:
```suggestion
			'Endpoint to get token required to view a user\'s profile details.'
```

File: health/middlewares.js (hunk 1 above)
Comment (thegreatercurve) [reply health/middlewares.js:3]: This is the dependency that the original authors of the repo use to get environment variables. I've added this to Vault

File: health/middlewares.js (hunk 2 above)
Comment (thegreatercurve) [reply health/middlewares.js]: Changed

File: health/middlewares.js (hunk 4 above)
Comment (thegreatercurve) [reply health/middlewares.js]: Changed

File: health/middlewares.js (hunk 5 above)
Comment (thegreatercurve) [reply health/middlewares.js]: Changed

File: health/middlewares.js (hunk 1 above)
Comment (binaryberry) [reply health/middlewares.js:3]: ok!

**Pull request merged**